### PR TITLE
[BotInterface] add current currency reference market values to portfolio

### PR DESCRIPTION
**Pull request by techfreaque**
This adds current reference market values for each holding for the telegram interface /portfolio

requires pull request in octobot commons

### CODE REVIEW

File: octobot_services/interfaces/util/portfolio.py [hunk 1]
```diff
@@ -109,6 +109,18 @@ def get_global_portfolio_currencies_amounts():
     return real_global_portfolio, simulated_global_portfolio
 
 
+def get_global_portfolio_currencies_values():
+    currencies_values = {}
+    for exchange in interfaces.get_exchange_managers():
+        this_currency_values = (
+            exchange.exchange_personal_data.portfolio_manager.portfolio_value_holder.current_crypto_currencies_values
```
Comment (Herklos) [inline octobot_services/interfaces/util/portfolio.py:116]: We try to not call directly internal logic but to create an api in octobot-trading. For example here, if it doesn't already exist, we have to create an api to request current crypto values from an exchange. This prevent having to change OctoBot-Services when modifying OctoBot-Trading internals.

Comment (techfreaque) [reply octobot_services/interfaces/util/portfolio.py:116]: Okay I'll do that

File: octobot_services/notifier/abstract_notifier.py [hunk 2]
```diff
@@ -103,9 +103,10 @@ async def _handle_order_notification(self, dict_order, linked_notification, orde
             self.previous_notifications_by_identifier[order_identifier] = notification
         else:
             is_simulated = trading_api.is_trader_simulated(exchange_manager)
-            if order_status is trading_enums.OrderStatus.CANCELED or \
+            if order_status is trading_enums.OrderStatus.EXPIRED or \
+                order_status is trading_enums.OrderStatus.CANCELED or \
                     (order_status is trading_enums.OrderStatus.CLOSED
-                     and dict_order[trading_enums.ExchangeConstantsOrderColumns.FILLED.value] == 0):
+                     and dict_order[trading_enums.ExchangeConstantsOrderColumns.FILLED_AMOUNT.value] == 0):
```
Comment (Herklos) [inline octobot_services/notifier/abstract_notifier.py:109]: Can you restore this change so we can merge before merging trading please?

**Review by Herklos — COMMENTED** [review]
👍

File: octobot_services/notifier/abstract_notifier.py (hunk 2 above)
Comment (techfreaque) [reply octobot_services/notifier/abstract_notifier.py]: Done, I moved it to a different PR

File: octobot_services/interfaces/util/portfolio.py [hunk 3]
```diff
@@ -109,6 +110,12 @@ def get_global_portfolio_currencies_amounts():
     return real_global_portfolio, simulated_global_portfolio
 
 
+def get_global_portfolio_currencies_values() -> dict:
+    return trading_api.get_global_portfolio_currencies_values(
```
Comment (GuillaumeDSM) [inline octobot_services/interfaces/util/portfolio.py:114]: 👍

File: octobot_services/interfaces/bots/abstract_bot_interface.py [hunk 4]
```diff
@@ -233,12 +235,17 @@ def get_command_sell_all(currency):
...
-        portfolios_string = f"{bold}{trader_str}{bold}Portfolio value : " \
-                            f"{bold}{pretty_printer.get_min_string_from_number(current_val)} {ref_market}{bold}" \
-                            f"{interfaces.EOL}"
-        portfolio_str = pretty_printer.global_portfolio_pretty_print(portfolio, markdown=markdown)
+        portfolios_string = (
+            f"{bold}{trader_str}{bold}Portfolio value : "
+            f"{bold}{pretty_printer.get_min_string_from_number(current_val)} {ref_market}{bold}"
+            f"{interfaces.EOL}"
+        )
+        portfolio_str = pretty_printer.global_portfolio_pretty_print(portfolio, currency_values, ref_market, markdown=markdown)
```
[the remark this answers is not in the document]
Comment (techfreaque) [reply octobot_services/interfaces/bots/abstract_bot_interface.py]: done, + it requires this PR https://github.com/Drakkar-Software/OctoBot/pull/2285

**Pull request merged**